### PR TITLE
Add MockApi for testing

### PR DESCRIPTION
**Pull request by bdunne**
@syncrou I think it's about time we start testing with real data.  There's more coming, but in this PR I covered the root collection classes.

I know there is some duplication and the `get` method needs to be smarter, but I wanted to keep this relatively small so it isn't impossible to review.  Let me know what you think.

I even found and fixed a bug in a spec which proves that the previous mocking wasn't adding much value for the `#initialize` tests.

### CODE REVIEW

**Comment by bdunne** [comment]
I know the PR looks huge, but most of the additions are in the mock instance data.

File: spec/support/mock_api.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+module AnsibleTowerClient
```
Comment (syncrou) [inline spec/support/mock_api.rb]: Do we want the `MockApi` available via the `AnsibleTowerClient` namespace? - It seems like maybe it should live under another namespace - possibly `AnsiblTowerClientTesting` or `AnsibleTowerClientMocks` - I guess in that case we could simply call the class `Api` without a collision.

Comment (bdunne) [inline spec/support/mock_api.rb]: I thought about this for a couple days and feel like it's better to keep all components of the gem inside the gem namespace.

Comment (syncrou) [inline spec/support/mock_api.rb]: I did more digging myself - and although I'd like to see our mocking and testing libraries under a separate namespace, we currently have the `FactoryHelper` under the `AnsibleTowerClient` namespace [here](https://github.com/ansible/ansible_tower_client_ruby/blob/master/spec/support/factory_helper.rb#L3-#L4).  So in the interest of consistency let's keep things consistent.

**Comment by bdunne** [comment]
@syncrou Yeah, I agree with your concerns.  I think we have the same problems with the existing test setup, so at least it's not getting worse.  I think there is a lot of room for improvement, but I think this allows for it.

**Comment by durandom** [comment]
For @syncrou valid concern:

aws-sdk-ruby does [stubbing / mocking](https://github.com/aws/aws-sdk-ruby/tree/082ea490f63225c159cce9639ccd293d2433044a/aws-sdk-core/lib/aws-sdk-core/stubbing) too. Plus they [store](https://github.com/aws/aws-sdk-ruby/tree/master/aws-sdk-core/apis/ec2/2016-11-15) their apis and example data in json files.

Maybe we could store the json repsonse of OPTIONS requests on https://ansible-tower.example.com/api/v1/projects/ in our repo and validate against that.

**Pull request merged**